webapp/src/main/java/com/box/l10n/mojito/service/tm/TMService.java

```diff
@@ -706,15 +705,61 @@ public String generateLocalized(
 
         logger.debug("Configuring pipeline for localized XLIFF generation");
 
+        BasePipelineStep translateStep = (BasePipelineStep) new TranslateStep(asset, repositoryLocale, InheritanceMode.USE_PARENT);
+        return generateLocalizedBase(asset, content, filterConfigIdOverride, bcp47Tag, translateStep);
+    }
+
+    /**
+     * Parses the given content and adds the pseudo localization for every text unit.
+     * Returns the pseudolocalized content.
+     * 
+     * @param asset The {@link Asset} used to get translations
+     * @param content The content to be localized
+     * @param repositoryLocale the repository locale used to fetch the
+     * translation. Also used for the output tag if outputBcp47tag is null.
+     * @param outputBcp47tag Optional, can be null. Allows to generate the file
+     * for a bcp47 tag that is different from the repository locale (which is
+     * still used to fetch the translations). This can be used to generate a
+     * file with tag "fr" even if the translations are stored with fr-FR
+     * repository locale.
+     * @return the localized asset
+     */
+    public String generatePseudoLocalized(
+            Asset asset,
+            String content,
+            RepositoryLocale repositoryLocale,
+            String outputBcp47tag,
+            FilterConfigIdOverride filterConfigIdOverride) {
+
+        String bcp47Tag;
+
+        if (outputBcp47tag == null) {
+            bcp47Tag = repositoryLocale.getLocale().getBcp47Tag();
```

Comment:
this whole block to identify the bcp47 tag should be in `generateLocalized` and not here